### PR TITLE
Editor: support go to implementation

### PR DESCRIPTION
**Pull request by LeonHuston**
This PR intends to add support for the LSP command `GoToImplementation` to the editor/project. The actual implementation of this feature piggy backs off of a lot of the work already done for the existing `GoToDefinition / GoToTypeDefinition` work. Which is why I didn't exactly add any new tests for this because functionally everything is the same apart from the underyling LSP request (which even has the same signature as the other ones) If I'm mis-reading the project and theres a spot i need to add tests for just let me know and I will.

Also notably I left out modifications to the keymap files as even though this functionality exists in both the VSCode and Jetbrains keymaps they overlap with existing default keymaps in this project. I wasn't sure if we just want to overwrite the behaviour of these keymaps for current users. Also that it looks like the VSCode keymaps uses `GoToImplementation` vs `GoToDefinition` for the same shortcut depending on LSP Capabilities and I wasn't sure if our current keymap crate has support for constraining keymaps based on LSP Context

Im very new to rust/this project but I'm very excited for this so just let me know if I did something janky and I will resolve :) 

Release Notes:

- Added "Go to implementation" support in editor.

### CODE REVIEW

**Comment by LeonHuston** [comment]
@cla-bot check

File: crates/rpc/proto/zed.proto [hunk 1]
```diff
@@ -48,145 +48,147 @@ message Envelope {
...
+        DeleteNotification delete_notification = 154;
+        MarkNotificationRead mark_notification_read = 155;
+        LspExtExpandMacro lsp_ext_expand_macro = 156;
+        LspExtExpandMacroResponse lsp_ext_expand_macro_response = 157;
+        SetRoomParticipantRole set_room_participant_role = 158;
+
+        UpdateUserChannels update_user_channels = 159; // current max
     }
 
-    reserved 158 to 161;
```
Comment (LeonHuston) [inline crates/rpc/proto/zed.proto:189]: I couldn't find anything about why we are reserving these specific Protobuf messages, I'm assuming they are just supposed to be filling the gap between `146 - 149` but wasn't sure. So just went with bumping this

**Review by osiewicz — CHANGES_REQUESTED** [review]
LGTM sans RPC message IDs. Once that's sorted out, we should be good to merge. :)

File: crates/rpc/proto/zed.proto (hunk 1 above)
Comment (osiewicz) [reply crates/rpc/proto/zed.proto:189]: Sweet, can we leave 158 to 161 as reserved though? I think there are still versions of Zed in the wild that can send/receive these messages, so 162 and 163 should be a safer bet. Also, that current max comment is a bit redundant, as the list is already sorted by id. Other than that this looks great!

Comment (LeonHuston) [reply crates/rpc/proto/zed.proto:189]: Done it :)

**Comment by osiewicz** [comment]
I've noticed that the item label for Implementations tab still says "Definitions for":
image

But I'll address that in a follow-up PR. Thanks for your contribution! =)

**Pull request merged**